WordPress/gutenberg
# Edit Post: Refactor More (Ellipsis) Menu

## Description
This PR adds a couple of refactorings to make More Menu extensible:
* Renames `EllipsisMenu` to `MoreMenu` to align with what was discussed on Slack some time ago.
* Removes div separators for the groups in the menu and replaces them with more flexible plain CSS solution.
* Moves `CopyContentButton` and hooks from `editor` folder to `editPost` folder.
* Makes all groups of menu items extensible by exposing filters that follow the same pattern: `editPost.MoreMenu.${ groupName )`.
* Extracts `MenuItemsChoice` component to be able to pass it down to the menu group.

## How to test?

Make sure all existing menu items display and work as before.

You can dynamically add new menu items by executing the following code:

```js
wp.hooks.addFilter( 'editPost.MoreMenu.editor', 'core/edit-post/editor-callback', ( items ) => [ ...items, 'Editor item' ] );
wp.hooks.addFilter( 'editPost.MoreMenu.settings', 'core/edit-post/editor-callback', ( items ) => [ ...items, 'Settings item' ] );
wp.hooks.addFilter( 'editPost.MoreMenu.tools', 'core/edit-post/editor-callback', ( items ) => [ ...items, 'Tools item' ] );
```

You might need to reopen More Menu to see the changes applied. We can fix that in the follow-up PR if we decide to have it dynamically re-render when new filters gets added or removed. It can be achieved by extracting most of the logic or even updating `withFilters` HOC: https://github.com/WordPress/gutenberg/blob/master/components/higher-order/with-filters/index.js#L26.

## Screenshots (jpeg or gifs if applicable):

More Menu with filters applied:

![screen shot 2018-02-20 at 12 57 27](https://user-images.githubusercontent.com/699132/36422726-a2f29a38-163d-11e8-80e7-ee92faf6b11a.png)

In the screenshot I added plain text, that’s why it doesn’t look perfect. We would rather expose a few components to consume there to make sure it looks properly.

## Checklist:
